### DISCUSSION TITLE
Troubleshooting Microsoft 365 Mail app

### DISCUSSION THREAD

**Discussion by mike-credisense** [datalust/seq-tickets#2155] [Help] [answered]
I've installed Seq.App.Mail.Microsoft365 to send outbound emails off alerts

I set up the app registration in azure and gave it the mail.send rights
I added an instance and copied all the details from the app registration, I wasn't sure if it needed a call back url? 

I added the email service to an alert that id receiving events and set a basic email to go out with default static text
The alert shows the emails are triggered but no email arrives.
How can I trouble shoot this in seq?, it's not showing in the logs....

**Comment by nblumhardt** [comment]
Hi Mike,

In the Events screen, if you pick an arbitrary event and expand it, and then choose "Send to app > (your Microsoft 365 mail app instance)" does the email arrive?

If not, are any errors logged from the app in that scenario?

If there's no error raised on the Seq side, you might find some useful info on the Microsoft 365 side, if there's a delivery log over there.

Another place to check - Settings > Diagnostics > Download a full diagnostic report. Checking the most recent events logged, at the very end of the report file, might reveal more info.

Hope this helps,
Nick

**Reply by mike-credisense** [reply]
Hi Nick

Looking at the app instance I set up I found **_View diagnostic events from this app instance_**  under the Title field
this shows the logs for the instance of the app. Looking in here showed an invalid user so I've change from email

testing again with the different user account I the error changes to: 
_Seq.Apps.SeqAppException: Microsoft Graph request failed (ErrorAccessDenied): Access is denied. Check credentials and try again._

This is my email I used for the from email but ultimately, I would like to it come from out support mail account. Is there any advice on the app registration to send as a specific mail account?

**Reply by mike-credisense** [reply, marked as answer]
Worked it out needed the **send as any user** permission and the grant permissions in azure

**Reply by nblumhardt** [reply]
Fantastic, thanks for sharing the solution
